### PR TITLE
Run tests with ipfs-api and go-ipfs again

### PR DESCRIPTION
**Pull request by haadcode**
This is a PR to make OrbitDB run tests with go-ipfs again (through js-ipfs-api) in addition to running them with js-ipfs.

This PR will:
- Remove obsolete/unused public API methods
- Run tests against both js-ipfs and go-ipfs
- Add write benchmark for using go-ipfs as the IPFS backend

### CODE REVIEW

**Comment by diasdavid** [comment]
👏🏽👏🏽👏🏽👏🏽👏🏽

File: test/docstore.test.js [hunk 1]
```diff
@@ -3,160 +3,169 @@
...
-      }
-      db = await orbitdb1.docstore(config.dbname, options)
+      if (ipfsd) 
+        await stopIpfs(ipfsd)
     })
 
-    afterEach(async () => {
-      await db.drop()
+    it('creates and opens a database', async () => {
+      db = await orbitdb1.docstore('first doc database')
```
Comment (thiagodelgado111) [inline test/docstore.test.js]: Duplicate?

Comment (haadcode) [reply test/docstore.test.js]: Good catch and it seems to be a problem with all store. They also don't seem to contain an assert for that test.

**Comment by haadcode** [comment]
I'm going to split this PR in two, because there's really two major things in it atm:
- Use ipfsd-ctl to create test IPFS instances
- Run tests against both js-ipfs and go-ipfs

The first part is to just update the tests to use ipfsd-ctl and the second part brings in everything we need to run with go-ipfs again.

**Comment by haadcode** [comment]
Closes #273

**Comment by haadcode** [comment]
Using ipfsd-ctl to run the tests is now a separate PR here https://github.com/orbitdb/orbit-db/pull/335

**Comment by haadcode** [comment]
This was now finally finished and in itself became a rather small PR compared to the original. The main work was done in the previous PRs (see commit log).

Going to merge this to master and keep testing it for some time before making a new release (0.20.0).

We now support go-ipfs again! 🎉 🚀 😄

**Pull request merged**